File: packages/contracts/src/enums.ts
export const CHANNEL_TYPES = [
  "web_chat",
  "whatsapp",
] as const;

export type ChannelType = (typeof CHANNEL_TYPES)[number];

export const CONVERSATION_STATUSES = [
  "active",
  "waiting",
  "completed",
  "error",
  "human_review_required",
] as const;

export type ConversationStatus = (typeof CONVERSATION_STATUSES)[number];

export const MESSAGE_DIRECTIONS = [
  "inbound",
  "outbound",
  "system",
] as const;

export type MessageDirection = (typeof MESSAGE_DIRECTIONS)[number];

export const MESSAGE_STATUSES = [
  "draft",
  "received",
  "accepted",
  "validating",
  "processing",
  "forwarded",
  "waiting",
  "completed",
  "error",
  "human_review_required",
] as const;

export type MessageStatus = (typeof MESSAGE_STATUSES)[number];

export const ATTACHMENT_STATUSES = [
  "received",
  "uploading",
  "uploaded",
  "validating",
  "validated",
  "rejected",
  "error",
] as const;

export type AttachmentStatus = (typeof ATTACHMENT_STATUSES)[number];

export const PROCESSING_EVENT_TYPES = [
  "conversation.created",
  "message.received",
  "attachment.upload.started",
  "attachment.upload.completed",
  "attachment.validation.started",
  "attachment.validation.completed",
  "processing.started",
  "actor.invoked",
  "actor.progress",
  "actor.completed",
  "actor.failed",
  "handoff.requested",
  "review.required",
  "response.partial",
  "response.final",
  "processing.completed",
] as const;

export type ProcessingEventType = (typeof PROCESSING_EVENT_TYPES)[number];

export const PROCESSING_STATUSES = [
  "pending",
  "running",
  "completed",
  "failed",
  "waiting",
  "human_review_required",
] as const;

export type ProcessingStatus = (typeof PROCESSING_STATUSES)[number];

export const REVIEW_TASK_STATUSES = [
  "open",
  "in_review",
  "resolved",
  "cancelled",
] as const;

export type ReviewTaskStatus = (typeof REVIEW_TASK_STATUSES)[number];

export const ARCHITECTURE_MODES = [
  "centralized_orchestration",
  "structured_workflow",
  "decentralized_swarm",
] as const;

export type ArchitectureMode = (typeof ARCHITECTURE_MODES)[number];

export const RUNTIME_MODES = [
  "mock",
  "real",
] as const;

export type RuntimeMode = (typeof RUNTIME_MODES)[number];

export const MOCK_ACTOR_NAMES = [
  "router_agent",
  "faq_agent",
  "stock_agent",
  "image_intake_agent",
  "supervisor_agent",
] as const;

export type MockActorName = (typeof MOCK_ACTOR_NAMES)[number];

File: apps/api/app/schemas/enums.py
from enum import StrEnum


class ChannelType(StrEnum):
    WEB_CHAT = "web_chat"
    WHATSAPP = "whatsapp"


class ConversationStatus(StrEnum):
    ACTIVE = "active"
    WAITING = "waiting"
    COMPLETED = "completed"
    ERROR = "error"
    HUMAN_REVIEW_REQUIRED = "human_review_required"


class MessageDirection(StrEnum):
    INBOUND = "inbound"
    OUTBOUND = "outbound"
    SYSTEM = "system"


class MessageStatus(StrEnum):
    DRAFT = "draft"
    RECEIVED = "received"
    ACCEPTED = "accepted"
    VALIDATING = "validating"
    PROCESSING = "processing"
    FORWARDED = "forwarded"
    WAITING = "waiting"
    COMPLETED = "completed"
    ERROR = "error"
    HUMAN_REVIEW_REQUIRED = "human_review_required"


class AttachmentStatus(StrEnum):
    RECEIVED = "received"
    UPLOADING = "uploading"
    UPLOADED = "uploaded"
    VALIDATING = "validating"
    VALIDATED = "validated"
    REJECTED = "rejected"
    ERROR = "error"


class ProcessingEventType(StrEnum):
    CONVERSATION_CREATED = "conversation.created"
    MESSAGE_RECEIVED = "message.received"
    ATTACHMENT_UPLOAD_STARTED = "attachment.upload.started"
    ATTACHMENT_UPLOAD_COMPLETED = "attachment.upload.completed"
    ATTACHMENT_VALIDATION_STARTED = "attachment.validation.started"
    ATTACHMENT_VALIDATION_COMPLETED = "attachment.validation.completed"
    PROCESSING_STARTED = "processing.started"
    ACTOR_INVOKED = "actor.invoked"
    ACTOR_PROGRESS = "actor.progress"
    ACTOR_COMPLETED = "actor.completed"
    ACTOR_FAILED = "actor.failed"
    HANDOFF_REQUESTED = "handoff.requested"
    REVIEW_REQUIRED = "review.required"
    RESPONSE_PARTIAL = "response.partial"
    RESPONSE_FINAL = "response.final"
    PROCESSING_COMPLETED = "processing.completed"


class ProcessingStatus(StrEnum):
    PENDING = "pending"
    RUNNING = "running"
    COMPLETED = "completed"
    FAILED = "failed"
    WAITING = "waiting"
    HUMAN_REVIEW_REQUIRED = "human_review_required"


class ReviewTaskStatus(StrEnum):
    OPEN = "open"
    IN_REVIEW = "in_review"
    RESOLVED = "resolved"
    CANCELLED = "cancelled"


class ArchitectureMode(StrEnum):
    CENTRALIZED_ORCHESTRATION = "centralized_orchestration"
    STRUCTURED_WORKFLOW = "structured_workflow"
    DECENTRALIZED_SWARM = "decentralized_swarm"


class RuntimeMode(StrEnum):
    MOCK = "mock"
    REAL = "real"


class RunStatus(StrEnum):
    PENDING = "pending"
    RUNNING = "running"
    COMPLETED = "completed"
    FAILED = "failed"
    CANCELLED = "cancelled"
    HUMAN_REVIEW_REQUIRED = "human_review_required"
